samples/alarmbot-es6-botframework-webchat/src/webChatAdapter.js

```diff
@@ -0,0 +1,25 @@
+const activitiesById = {};
+
+export class WebChatAdapter {
+    post(activities) {
+        activitiesById[activities[0].replyToId] = activities;
+        activities.forEach(activity => Object.assign(activity, {id: activity.replyToId, from: {id: "The Bot"}}));
+        return Promise.resolve(activities);
+    }
+
+    onReceive(activity) {
+        // overwritten by bot
+    }
+
```

Comment:
I'm assuming the reason you chose to use a function here versus a property is because in ES6 you can't show that its typed to take a function.  I wonder if it would be cleaner use a setter here. I just filled an issue with TypeScript that there's currently no way to express in an interface that something is writeonly.  This is a classic "weak reference" pattern.